### ISSUE TITLE
Cannot tab in react-testing-library

### ISSUE THREAD

**Issue by dannyharding10** [reactjs/react-modal#919]
### Summary:
When writing tests using react-testing-library, tabbing does not work properly when inside of a react-modal.

### Steps to reproduce:

 1. Render a Modal with some tabbable elements
 2. Try to tab between elements using (@testing-library/user-event)[https://testing-library.com/docs/ecosystem-user-event/] (userEvent.tab(\))[https://testing-library.com/docs/ecosystem-user-event/#tabshift-focustrap]
 3. The focused element does not change 😢

### Expected behavior:
The focused element changes when calling `userEvent.tab()` inside of a react-modal.

### Link to example of issue:
https://codesandbox.io/s/xenodochial-mestorf-qx6j8?file=/src/__tests__/index.js

### Additional notes:
I'm aware the issue could be in the (user-event)[https://testing-library.com/docs/ecosystem-user-event/] library, so I made a companion issue over there (clicky)[]. I'm also thinking I could just need more understanding of how the focus trapping is working in react-modal / user-event so I can get them to play together nicely.

**Comment by diasbruno** [comment]
Hey @dannyharding10, in react-modal we try not to change the browser's default behavior of tabbing. There is only one case we need to intervene and it's for radio buttons on Safari. Other than that we find the first and last "tabbable" elements to try trapping the focus on the modal.

**Comment by dannyharding10** [comment]
@diasbruno interesting, thanks for the tip. Are you saying that you don't believe the issue would be in react-modal because react-modal doesn't really do anything to interfere with tabbing?

**Comment by diasbruno** [comment]
Yep, but not with 100%, because we do intervene...we have some tests [to back it up.](https://github.com/reactjs/react-modal/blob/master/specs/Modal.helpers.spec.js) 😂 

Let me know if you can spot something wrong on our side.

**Comment by dannyharding10** [comment]
Ok thank you, I will be sure to report back here on whatever happens

**Comment by phyrwork** [comment]
We have the same issue and observe that issue seems to be introduced between 3.15.1 and 3.16.1.

[Code sandbox reproducing issue](https://codesandbox.io/s/exciting-glitter-6pgrjw) and [original discussion](https://github.com/testing-library/user-event/issues/820#issuecomment-1755529982) from user-event.
